Add a flag to HNS Load Balancer policy to hint HNS about the VIP being an External IP

princepereira opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind bug


#### What this PR does / why we need it:
For Service of Type Load Balancer and External Traffic Policy set to Local add a new Portmapping flag to the HNS Load Balancer policy which ensures that once a connection is directed to an endpoint, it continues to go to that endpoint despite changes in the list of endpoints backing the service.

<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Fixing issue in Winkernel Proxier - Unexpected active TCP connection drops while horizontally scaling the endpoints for a LoadBalancer Service with Internal Traffic Policy: Local
```

#### Which issue(s) this PR fixes:
#113842

princepereira committed:
Introducing LoadbalancerPortMapping flags for VipExternalIP

sbangari commented:
/ok-to-test

princepereira commented:
/cc @jsturtevant 
/cc @daschott 
/cc @sbangari 

Hi @jsurtevant, @daschott , @sbangari ,

Please review the code when you get some time.

Regards,
Prince

daschott reviewed (changes requested):
Need to add externalVIP to loadBalancerflags for load balancer ingress ip as well:
https://github.com/kubernetes/kubernetes/pull/113742/files#diff-03419cd19861179be871687a3763ba4440dc39fe688032ecb8e2cbd0efd6f61dL1377

princepereira commented:
> Need to add externalVIP to loadBalancerflags for load balancer ingress ip as well: https://github.com/kubernetes/kubernetes/pull/113742/files#diff-03419cd19861179be871687a3763ba4440dc39fe688032ecb8e2cbd0efd6f61dL1377

Thanks @daschott  . Addressed the comments.

princepereira commented:
/test pull-kubernetes-e2e-capz-windows-containerd

sbangari commented:
/milestone v1.27

sbangari commented:
/lgtm
/approve

princepereira commented:
/release-note-none

princepereira commented:
/test pull-kubernetes-e2e-gce-ubuntu-containerd

princepereira commented:
/test pull-kubernetes-e2e-gce-ubuntu-containerd

sbangari commented:
/lgtm

marosset commented:
/milestone v1.27

sbangari commented:
/lgtm

princepereira commented:
/test pull-kubernetes-integration
